### DISCUSSION TITLE
Changing the title of a PR check action in the same PR causes mass confusion

### DISCUSSION THREAD

**Discussion by chriskuech** [community/community#13792] [Pull Requests]
# Repro
1. Create a repo with branch protection and a mandatory GitHub Action check in the branch policy.
2. Create a pull request to the protected branch that changes the name of the required GitHub Action.

# Expected
* Some kind of indication that branch protection doesn't know what to do, and instructions on ways to resolve.
* Don't use the display name to identify actions.  If the action is required, allow providing an immutable ID.

# Observed
The required Action is stuck in the following state:
<img width="854" alt="image" src="https://user-images.githubusercontent.com/3680671/160675678-a4d2ddb6-3fad-4789-bb8a-fa44a4011d0a.png">

**Comment by tobylane** [comment]
Also a problem when the check is renamed (outside of the PR) after the PR is opened and before another commit is pushed to the PR.
